src/dodal/devices/eiger.py

```diff
@@ -85,7 +91,7 @@ def stage(self):
         LOGGER.info("Waiting on parameter callbacks")
         status.wait(self.STALE_PARAMS_TIMEOUT)
 
-        self.arm_detector()
+        return self.arm_detector()
```

Comment:
Must: With this now being asynchronous it no longer conforms to the interface for `stage` so we should rename it.